### PR TITLE
feat: add the reading page

### PR DESCRIPTION
**Pull request by NyAndoMayah**
Not perfectly responsive yet (for mobile browser)

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -17,6 +17,7 @@ module.exports = {
   parser: "@typescript-eslint/parser",
   plugins: ["react-refresh"],
   rules: {
+    "prettier/prettier": ["error", {endOfLine: "auto"}],
```
Comment (YumeT023) [inline .eslintrc.cjs]: Remove, prettier formats the code and change it to line feed

File: src/lib/get_reading_time.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// theory : https://stackoverflow.com/questions/4216752/anyone-having-any-leads-on-a-reading-time-algorithm
+export const calculateReadTime = (text: string) => {
```
Comment (YumeT023) [inline src/lib/get_reading_time.ts]: move this to features/post/util.ts

File: src/main.tsx [hunk 3]
```diff
@@ -6,6 +6,7 @@ import {LoginPage} from "@/pages/auth";
 import {Authenticated} from "@/features/auth";
 import {AuthProvider} from "@/services/auth_provider.ts";
 import "./index.css";
+import {Post} from "./pages/post";
```
Comment (YumeT023) [inline src/main.tsx]: use @/ path alias instead,
and follow the imports order

File: src/pages/post/Post.tsx [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+} from "@/components/ui/navigation-menu";
+import {Badge} from "@/components/ui/badge";
+import {Link} from "react-router-dom";
+import {Icon} from "@iconify/react";
+import image from "../../assets/daisy_fields.jpg";
+import blankUserProfile from "../../assets/noun-user-picture.svg";
+import {Avatar, AvatarFallback, AvatarImage} from "@/components/ui/avatar";
+import {posts} from "./data/posts";
+import {users} from "./data/user";
+import {calculateReadTime} from "../../lib/get_reading_time";
```
Comment (YumeT023) [inline src/pages/post/Post.tsx]: use @/ import alias ... get rid of ../../ ... mental mapping

File: src/pages/post/Post.tsx [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+  NavigationMenu,
+  NavigationMenuItem,
+  NavigationMenuLink,
+  NavigationMenuList,
+  navigationMenuTriggerStyle,
+} from "@/components/ui/navigation-menu";
+import {Badge} from "@/components/ui/badge";
+import {Link} from "react-router-dom";
+import {Icon} from "@iconify/react";
+import image from "../../assets/daisy_fields.jpg";
```
Comment (YumeT023) [inline src/pages/post/Post.tsx]: rene image to concrete name 
and use import alias @/

File: src/pages/post/Post.tsx [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+import blankUserProfile from "../../assets/noun-user-picture.svg";
+import {Avatar, AvatarFallback, AvatarImage} from "@/components/ui/avatar";
+import {posts} from "./data/posts";
```
Comment (YumeT023) [inline src/pages/post/Post.tsx]: TODO: replace static data with cypress tests

File: src/pages/post/Post.tsx [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+import {users} from "./data/user";
+import {calculateReadTime} from "../../lib/get_reading_time";
+
+const BADGES = ["lorem", "ipsum", "hello"];
+
+const NavBar: React.FC = () => {
```
Comment (YumeT023) [inline src/pages/post/Post.tsx]: use named import {FC} instead of importing the whole React namespace

File: src/pages/post/Post.tsx [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+  return (
+    <div
+      id="blogify-menu"
```
Comment (YumeT023) [inline src/pages/post/Post.tsx]: what is this id for ?

File: src/pages/post/Post.tsx [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+            fontOpticalSizing: "auto",
+            fontWeight: 618,
+            fontStyle: "normal",
+          }}
+        >
+          {currentPost?.title}
+        </p>
+      </div>
+      <div className="flex w-screen justify-center">
+        <div id="blog-info" className="grid w-[500px] grid-cols-3">
```
Comment (YumeT023) [inline src/pages/post/Post.tsx]: what are these ids for?

File: src/pages/post/Post.tsx [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+          </div>
+          <div
+            id="read-duration-info"
+            className="flex items-center justify-center"
+          >
+            <Icon
+              icon="material-symbols-light:nest-clock-farsight-analog-outline"
+              className="text-2xl"
+            />
+            <span className="mx-1">{`${calculateReadTime(currentPost?.content || "").minutes} min read`}</span>
```
Comment (YumeT023) [inline src/pages/post/Post.tsx]: add default value on text param of calc... fn

File: src/pages/post/Post.tsx [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+                <li>menu menu menu menu</li>
+              </ul>
+            </div>
+          </div>
+          <div id="post-container" className="col-span-6 p-4">
+            <div className="mx-10">{currentPost?.content}</div>
+            <div id="tags-container" className="mx-10 flex w-full py-10">
+              <span className="mr-2">Tags : </span>
+              <div className="flex justify-evenly">
+                {BADGES.map((badge) => (
```
Comment (YumeT023) [inline src/pages/post/Post.tsx]: missing key prop,
avoid unnecessary loops, write them directly

File: src/pages/post/Post.tsx [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+                  <Badge className="mx-1">{badge}</Badge>
+                ))}
+              </div>
+            </div>
+          </div>
+        </div>
+        <div className="col-span-1 h-[300px]"></div>
```
Comment (YumeT023) [inline src/pages/post/Post.tsx]: avoid absolute unit like "px", instead use tailwindcss relative ones or "rem"

File: src/pages/post/data/posts.ts [hunk 13]
```diff
@@ -0,0 +1,53 @@
+export enum Status {
+  ARCHIVED,
+  ENABLED,
+  DRAFT,
+}
+
+export interface Post {
```
Comment (YumeT023) [inline src/pages/post/data/posts.ts]: use the models from @/services/gen from now on

File: src/pages/post/index.tsx [hunk 14]
```diff
@@ -0,0 +1 @@
+export {Post} from "./Post";
```
Comment (YumeT023) [inline src/pages/post/index.tsx]: export * front...

ensure there is no private utility/cmp that are exported

File: src/pages/post/Post.tsx [hunk 15]
```diff
@@ -0,0 +1,222 @@
...
+  return (
+    <div className=" overflow-hidden bg-white">
+      <NavBar />
+      <div
+        id="post-title"
+        className="mb-4 mt-40 flex w-screen justify-center p-11 md:mt-20"
+      >
+        <p
+          className="text-6xl"
+          style={{
```
Comment (YumeT023) [inline src/pages/post/Post.tsx]: Find better way to set the font family

**Review by YumeT023 — CHANGES_REQUESTED** [review]

**Review by YumeT023 — CHANGES_REQUESTED** [review]
If you rebased `preprod` correctly, YOu should have shadcn-ui generated components in the _src/components/shadcn-ui_

**Comment by YumeT023** [comment]
image

I really like how it turned out :D

Some improvement:
- The title / menu / the post content is kinda off, like the title is centered, but the content itself isn't, the menu is at left ...
  - But leave it as is, @JrManjato will tell you how to make it prettier and all the improvements that need to be done when the code review itself is done.
- Remove the header shadow and replace it with a solid (pencil-like) line instead
- Extend the Banner more, look at it as how youtuber look at their youtube video thumbnail
e.g
image

File: src/pages/post/Post.tsx (hunk 10 above)
Comment (NyAndoMayah) [reply src/pages/post/Post.tsx]: already calculated by the calculateReadTime function

Comment (YumeT023) [reply src/pages/post/Post.tsx]: function calculateReadTime(text = "") 
to always handle nullish input to calculateReadTime by default

File: src/main.tsx [hunk 16]
```diff
@@ -1,10 +1,14 @@
 import React from "react";
+
```
Comment (YumeT023) [inline src/main.tsx]: no space needed

File: src/main.tsx [hunk 17]
```diff
@@ -34,6 +38,10 @@ const ROUTER = createBrowserRouter([
       </Authenticated>
     ),
   },
+  {
+    path: "/posts/:id",
+    element: <Post />,
```
Comment (YumeT023) [inline src/main.tsx]: follow page naming convention, suffix with Page
Post -> **PostPage**

File: src/pages/post/Post.tsx [hunk 18]
```diff
@@ -0,0 +1,193 @@
+import {FC} from "react";
+
```
Comment (YumeT023) [inline src/pages/post/Post.tsx]: too much space,

instead group them as: lib_imports, local_imports

File: src/pages/post/Post.tsx [hunk 19]
```diff
@@ -0,0 +1,193 @@
...
+  NavigationMenuLink,
+  NavigationMenuList,
+  navigationMenuTriggerStyle,
+} from "@/components/shadcn-ui/navigation-menu";
+
+// TODO: replace static data with cypress tests
+import {users} from "./data/user";
+import {posts} from "./data/posts";
+
+import banner_image from "@/assets/daisy_fields.jpg";
```
Comment (YumeT023) [inline src/pages/post/Post.tsx]: consistent naming convention, 
use camelCase

File: src/pages/post/Post.tsx [hunk 20]
```diff
@@ -0,0 +1,193 @@
...
+  return (
+    <div className=" overflow-hidden bg-white">
+      <NavBar />
+      <div className="mb-4 mt-40 flex w-screen justify-center p-11 md:mt-20">
+        <p className="font-optical-sizing-auto normal font-title text-6xl font-bold">
+          {currentPost?.title}
+        </p>
+      </div>
+      <div className="flex w-screen justify-center">
+        <div className="grid w-[500px] grid-cols-3">
```
Comment (YumeT023) [inline src/pages/post/Post.tsx]: do not use absolute unit 'px'

File: src/pages/post/Post.tsx (hunk 10 above)
Comment (YumeT023) [reply src/pages/post/Post.tsx]: calculateReadTime -> calculateReadDuration

File: src/features/post/utils.ts [hunk 21]
```diff
@@ -0,0 +1,15 @@
+// theory : https://stackoverflow.com/questions/4216752/anyone-having-any-leads-on-a-reading-time-algorithm
+export const calculateReadTime = (text: string) => {
```
Comment (YumeT023) [inline src/features/post/utils.ts]: add default text
```js
export const calculateReadTime = (text = "") => {
```

File: src/main.tsx [hunk 22]
```diff
@@ -1,10 +1,14 @@
 import React from "react";
+
+import {createBrowserRouter, RouterProvider} from "react-router-dom";
```
Comment (YumeT023) [inline src/main.tsx:2]: order your imports

File: src/pages/post/Post.tsx [hunk 23]
```diff
@@ -0,0 +1,193 @@
...
+} from "@/components/shadcn-ui/navigation-menu";
+
+// TODO: replace static data with cypress tests
+import {users} from "./data/user";
+import {posts} from "./data/posts";
+
+import banner_image from "@/assets/daisy_fields.jpg";
+import blankUserProfile from "@/assets/noun-user-picture.svg";
+
+const NavBar: FC = () => {
```
Comment (YumeT023) [inline src/pages/post/Post.tsx]: TODO: move else where

e.g: layout dir

File: src/pages/post/Post.tsx [hunk 24]
```diff
@@ -0,0 +1,193 @@
...
+                "No date provided"}
+            </span>
+          </div>
+        </div>
+      </div>
+      <div className="my-5 grid grid-cols-8">
+        <div className="col-span-1 h-[18.75rem]"></div>
+        <div className="col-span-6 grid grid-cols-8 gap-5">
+          <div className="col-span-8 h-[30rem]">
+            <img src={banner_image} className="h-full w-full object-cover" />
```
Comment (YumeT023) [inline src/pages/post/Post.tsx]: not large enough

example of banner img: https://www.freecodecamp.org/news/the-biggest-changes-in-javascript-this-year/?ref=dailydev

File: src/pages/post/Post.tsx [hunk 25]
```diff
@@ -0,0 +1,193 @@
...
+            <img
+              src={blankUserProfile}
+              className="m-auto h-[200px] w-[200px] rounded-md bg-white object-cover"
+            />
+          </div>
+          <div className="container col-span-6 flex flex-col justify-center">
+            <Link
+              to=""
+              className="mb-5 text-left font-title text-2xl hover:text-slate-700 focus:text-slate-200 active:font-semibold"
+            >{`${postAuthor?.first_name} ${postAuthor?.last_name}`}</Link>
```
Comment (YumeT023) [inline src/pages/post/Post.tsx]: no need to wrap in templat literal
```js
>{postAuthor.first_name} {postAuthor.last_name}</
```

**Review by YumeT023 — CHANGES_REQUESTED** [review]

**Comment by YumeT023** [comment]
See the diff

actual:
image

expected:
image
(almost invisible but adds lot of modernity)

**Comment by YumeT023** [comment]
Add vertical _solid line_ between menu and main content

image

**Comment by YumeT023** [comment]
* Missing tests (check codecov coverage reporting)
  * e2e is the most important (add unit if needed, for the readingDuration calculation for e.g)

File: src/pages/post/PostPage.tsx [hunk 26]
```diff
@@ -0,0 +1,195 @@
+import {FC} from "react";
+import {Link} from "react-router-dom";
+
+import {Icon} from "@iconify/react";
+import {Badge} from "@/components/shadcn-ui/badge";
+import {Button} from "@/components/shadcn-ui/button";
+import {calculateReadDuration} from "@/features/post/utils";
+import {
```
Comment (YumeT023) [inline src/pages/post/PostPage.tsx]: order imports

components
features
services

File: src/pages/post/PostPage.tsx [hunk 27]
```diff
@@ -0,0 +1,195 @@
...
+            />
+            <span className="mx-1">{`${calculateReadDuration(currentPost?.content).minutes} min read`}</span>
+          </div>
+          <div className="flex items-center justify-center">
+            <Icon
+              icon="material-symbols-light:calendar-month-outline"
+              className="text-2xl"
+            />
+            <span className="mx-1">
+              {currentPost?.creation_datetime?.toLocaleDateString() ||
```
Comment (YumeT023) [inline src/pages/post/PostPage.tsx]: TODO: relative datetime (like: 1 week ago) for later

File: src/pages/post/PostPage.tsx [hunk 28]
```diff
@@ -0,0 +1,195 @@
...
+            <span className="mx-1">
+              by <strong>{postAuthor.first_name}</strong>
+            </span>
+          </div>
+          <div className="flex items-center justify-center">
+            <Icon
+              icon="material-symbols-light:nest-clock-farsight-analog-outline"
+              className="text-2xl"
+            />
+            <span className="mx-1">{`${calculateReadDuration(currentPost?.content).minutes} min read`}</span>
```
Comment (YumeT023) [inline src/pages/post/PostPage.tsx]: unnecessary
```js
>{calculateReadDuration(currentPost?.content).minutes} min read</span...
```

**Review by YumeT023 — CHANGES_REQUESTED** [review]

**Pull request merged**